src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java

```diff
@@ -32,30 +36,8 @@ public GuiFluidicCompressor(InventoryPlayer inventoryPlayer, TileFluidicCompress
 	}
 
 	@Override
-	protected void mouseClicked(int mouseX, int mouseY, int mouseButton) {
-		super.mouseClicked(mouseX, mouseY, mouseButton);
-		int mX = mouseX - guiLeft;
-		int mY = mouseY - guiTop;
-		if (mX >= 20 && mX <= 39 && mY >= 25 && mY <= 41 && !fluidicCompressor.fill) {
-			fluidicCompressor.fill = true;
-			PacketHandler.instance.sendToServer(new MessageFluidicCompressorA(true, fluidicCompressor));
-		}
-		if (mX >= 20 && mX <= 39 && mY >= 45 && mY <= 61 && fluidicCompressor.fill) {
-			fluidicCompressor.fill = false;
-			PacketHandler.instance.sendToServer(new MessageFluidicCompressorA(false, fluidicCompressor));
-		}
-
-	}
-
-	@Override
 	public void drawBackgroundPreWidgets(float f, int x, int y) {
-		if (fluidicCompressor.fill) {
-			drawTexturedModalRect(guiLeft + 20, guiTop + 25, 195, 83, 19, 16);
-			drawTexturedModalRect(guiLeft + 20, guiTop + 45, 176, 99, 19, 16);
-		} else {
-			drawTexturedModalRect(guiLeft + 20, guiTop + 45, 195, 99, 19, 16);
-			drawTexturedModalRect(guiLeft + 20, guiTop + 25, 176, 83, 19, 16);
-		}
+		drawTexturedModalRect(guiLeft + 75, guiTop + 58 - (fluidicCompressor.fill ? 39 : 0), 195, 99 - (fluidicCompressor.fill ? 16 : 0), 25, 9);
```

Comment:
That would be very specific for a widget. In fact it does the same as the progress bar.